core/modules/simpletest/simpletest.module

```diff
@@ -394,9 +394,18 @@ function simpletest_test_get_all() {
                 continue;
               }
             }
+            // Verify that the test file is present in the same directory as
+            // specified in the .tests.info file.
             if (empty($error)) {
-            // If the format passed, build the rest of the groups information.
               $info['file path'] = dirname($file->uri);
+              $test_file_full = BACKDROP_ROOT . '/' . $info['file path'] . '/' . $info['file'];
+              if (!file_exists($test_file_full)) {
+                $error = t('File %uri specifies a non-existent file %test_file_full. Please verify that the test file exists in the location specified in %uri.', array('%test_file_full' => $test_file_full));
```

Comment:
I just realized, that this happens in a loop (`foreach ($classes as $class => $info)`). Doesn't that mean that if several paths are wrong, only the last file gets mentioned?

Reply:
In GHA tests, only the first .tests.info file will be mentioned. In the UI, each bad .tests.info file will be mentioned. However, only the first (not last) bad file path in a .tests.info file will be mentioned. Hopefully, this alerts the user to go examine their .tests.info file if they've done the same error in multiple places within it.

The reason for bailing out when an error is found, rather than, say, accumulating a list of errors, is that for some types of malformation, once the parser gets confused, it generates a whole slew of gibberish errors, so rather than stuffing watchdog (and the messages area) with all of that where only the first error in the .tests.info file was really relevant, we just report the first to let the person know they need to go take a look.

Reply:
Makes sense, many thanks for the explanation. Consider this question as resolved.